frontend/server/src/DAO/Problems.php

```diff
@@ -152,37 +152,86 @@ class Problems extends \OmegaUp\DAO\Base\Problems {
                 [$programmingLanguage],
             ];
         }
-        if (!is_null($difficultyRange) && count($difficultyRange) === 2) {
-            $conditions = 'p.difficulty >= ? AND p.difficulty <= ?';
-            if ($difficultyRange[0] === 0) {
-                $conditions = '(p.difficulty IS NULL OR (p.difficulty >= ? AND p.difficulty <= ?))';
-            }
 
-            $clauses[] = [
-                $conditions,
-                $difficultyRange,
-            ];
+        $difficulties = ['easy', 'medium', 'hard'];
+
+        // Convert the difficulty in text form to a range
+        if (is_null($difficultyRange)) {
+            $difficultyRange = [];
+            if (in_array($difficulty, $difficulties, true)) {
+                if ($difficulty === $difficulties[0]) {
+                    $difficultyRange[] = 0;
+                    $difficultyRange[] = 1;
+                } elseif ($difficulty === $difficulties[1]) {
+                    $difficultyRange[] = 2;
+                    $difficultyRange[] = 2;
+                } else {
```

Comment:
```suggestion
                } else if ($difficulty === 'hard') {
```

así se puede eliminar L156 y L161, porque ya está exhaustivamente verificado que las dificultades son algunas de esas tres.

o mejor aún: se puede usar un `switch` por consistencia con los de abajo?